### PR TITLE
Issue #3395303: Require Drupal 10 to make Open Social Drupal 10 ready

### PR DESCRIPTION
**Pull request by ronaldtebrake**
## Problem
Last part of our Drupal 10 efforts is to require Drupal 10 in our composer.json. 
Below you can find some of the interesting aspects from the [Drupal 10 release notes](https://www.drupal.org/project/drupal/releases/10.0.0) that applied to changes within our profile, please read the release notes to ensure other aspects are not touching your installation.

### Removed modules and themes

In this case we have moved to the contrib modules away from Core color and Ckeditor.
#3581 
#3579 

& according to https://www.drupal.org/node/3156376
We have fixed our `social_mentions` from `drupal/jquery_ui` and `drupal/jquery_ui_autocomplete` libraries 
[in this commit](https://github.com/goalgorilla/open_social/pull/3578/commits/5902697bcd7e8764626cfb68c631e8c317a00b5b) 

### Sites must update to at least Drupal 9.4.4 before upgrading to Drupal 10
This means we have to change the way our behat flow tests our update path, we can't run it from Open Social 10 anymore, we opted to run it from our last supported version (11.9.x) for the time being.

### Platform requirements changes
We're requiring Drupal 10 and PHP 8.1 for Open Social 12.x, so we can remove any PHP 8.0 checks (and lower) 
[Drupal 10 requires PHP 8.1 or higher](https://www.drupal.org/about/core/blog/drupal-10-will-require-php-81).

### Composer requirements
Drupal 10 [recommends Composer version 2.3.6 or higher](https://www.drupal.org/docs/system-requirements/composer-requirements), which is required for compatibility with PHP 8.2 and the forthcoming Automatic Updates feature. 

In Open Social we don't have a required version, but make sure to update yours!

> To update your host's version of Composer, run:
> `composer self-update`
>
> You can roll back to the previous version at any time by using: 
> `composer self-update --rollback`
>
> Update to a specific version with:
> `composer self-update 2.3.6`

Which meant we have quite some PHPStan related bug fixes as part of this PR.

[So we've opted to migrate the one from core for the time being](https://github.com/goalgorilla/open_social/pull/3578/commits/769a375e27b700c245b7b1177a2d32b2c494d758), necessary for our Emoji picker and we can move to the recommended successor `floating-ui` later one.

### Other changes affecting us

https://www.drupal.org/node/3187082
https://www.drupal.org/node/3129492

------- 

## Issue tracker
https://www.drupal.org/project/social/issues/3395303

## How to test
Full regression check of Open Social's features.

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Release notes
We are now Drupal 10 ready 🎉

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -46,26 +46,24 @@
...
+                "Issue #2107455: Image field default value not shown when upload destination set to private file storage": "https://www.drupal.org/files/issues/2022-10-27/2107455-79.patch",
                 "Issue #3052115: Mark an entity as 'syncing' during a migration update": "https://www.drupal.org/files/issues/2023-02-01/3052115-59.patch",
                 "Issue #3332546: CommentSelection::entityQueryAlter() fails on validate when referencing entity is not a comment": "https://www.drupal.org/files/issues/2023-01-11/3332546-6-comment_selection_entityqueryalter.patch",
                 "2924783 - Fatal error on entity autocomplete widget if entity label contains parentheses": "https://www.drupal.org/files/issues/2021-04-18/2924783-18.patch",
-                "States API doesn't work with multiple select fields": "https://www.drupal.org/files/issues/2023-06-27/drupal-n1149078-172-95x.patch"
+                "States API doesn't work with multiple select fields": "https://git.drupalcode.org/project/drupal/-/merge_requests/3805.patch"
```
Comment (ronaldtebrake) [inline composer.json:66]: Make this a non rolling target

File: composer.json [hunk 2]
```diff
@@ -124,8 +122,9 @@
         "drupal/block_field": "1.0.0-rc4",
         "drupal/config_modify": "^1",
         "drupal/config_update": "2.0.0-alpha3",
-        "drupal/core": "~9.5",
-        "drupal/core-composer-scaffold": "~9.5.3",
+        "drupal/core": "~10.0.0",
+        "drupal/core-composer-scaffold": "~10.0.0",
+        "drupal/color": "1.0.3",
```
Comment (ronaldtebrake) [inline composer.json:127]: Remove this, as we can do that in a separate PR

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

File: modules/social_features/social_post/modules/social_post_album/src/Element/SocialPostAlbumManagedFile.php
Comment (robertragas) [inline modules/social_features/social_post/modules/social_post_album/src/Element/SocialPostAlbumManagedFile.php:1]: This change causes
`InvalidArgumentException: in Drupal\social_post_album\Element\SocialPostAlbumManagedFile::uploadAjaxCallback() (line 28 of /var/www/html/profiles/contrib/social/modules/social_features/social_post/modules/social_post_album/src/Element/SocialPostAlbumManagedFile.php).`

It happens when you try to add an image to your post on the /stream

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

File: modules/social_features/social_core/social_core.install
Comment (robertragas) [inline modules/social_features/social_core/social_core.install:1]: Even though this looks good to me, i'm not sure why this is suddenly needed? Did we handle this some other way in Drupal 9?

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

File: modules/social_features/social_core/social_core.install
Comment (ronaldtebrake) [reply modules/social_features/social_core/social_core.install:1]: You can see it in the commit message; 

Database drivers are now moved in to a core module, we need to enable it. https://www.drupal.org/node/3129492

Otherwise we run in to a:

> The current database driver is provided by the module: /mysql/. The module is
> currently not enabled. You should immediately enable [1] the module.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

File: modules/social_features/social_profile/modules/social_profile_privacy/social_profile_privacy.module [hunk 3]
```diff
@@ -310,6 +310,7 @@ function _social_profile_privacy_fields_submit($form, FormStateInterface $form_s
 
   $profiles = \Drupal::entityQuery('profile')
     ->condition('uid', $uid)
+    ->accessCheck(FALSE)
```
Comment (robertragas) [inline modules/social_features/social_profile/modules/social_profile_privacy/social_profile_privacy.module:313]: Shouldn't this be TRUE? It's a form submit where it's triggered by the user himself.

Although I guess even if you can manipulate the post the only result is that you can force a cache invalidate for other profiles.

Comment (ronaldtebrake) [reply modules/social_features/social_profile/modules/social_profile_privacy/social_profile_privacy.module]: Seems like that wouldn't hurt, eventhough I thought it might make sense that a user actually saving the user entity might not have access to the user profile for certain cases I don't think those ever we're part of our use case (as the default is TRUE and that has been running for a while now)

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3578-zdqb9mxmyxkmj00i6eotaqbkfqivt3bk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/653bd38ac5102c775686066c

**Review by robertragas — APPROVED** [review]
